packages/lib-classifier/src/store/markings/LineStore.js

```diff
@@ -0,0 +1,97 @@
+import { addDisposer, getRoot, isValidReference, types } from 'mobx-state-tree'
+import { autorun } from 'mobx'
+import { filter, skip } from 'rxjs/operators'
+
+import Mark from './Mark'
+
+const Line = types
+  .model('LineStore', {
+    coordinatesArray: types.maybeNull(
+      types.array(types.model({
```

Comment:
Sounds good regarding `maybeNull`, I'm updating accordingly.

The thought process behind a `coordinatesArray` and `coordinates` view was so that all markings would store coordinates similarly (`coordinatesArray`), but have a unique view. I also found it kept the actions more straightforward; as opposed to manipulating an object (like `{ x1: number, y1: number, x2: number, y2: number }` which would be different for each marking) in the action that sets the most recent event coordinates, this keeps the set coordinates action to just pushing the latest coordinates. But I'm not confident this is the optimal approach, just initial attempt.

I'm not sure how editing a marking will fit into this structure.

Reply:
Attempted to refactor to set coordinates directly on model with `applyPatch` in `setCoordinates` action, console logged and confirmed patches were firing and everything looked to be functioning as expected, but in DOM the point/line component wasn't visually updating on pointer moves. It'd show point/line on initial pointer down, then if dragged mouse around nothing, but on pointer up the point or line would show as expected. Didn't dig too much into in light of upcoming drawing reassessment.

Reply:
I don't think you need to be updating the store until drawing is complete. At least, I don't think we're interested in observing mark changes while you're still drawing. Otherwise, I think that just leads us back into the performance issues that PFE has when the annotation is updated on every change to a mark.

I'd suggest rendering the SVG shapes from components and paths stored in component state, then updating the store with the final values when the pointer lifts.

Reply:
Storing in local component state definitely was the original plan, but it shifted to the store due to the attempt to integrate with the rx.js stream and the stream was initialized in the store... I'm not sure if there was any other complications that led to us shifting to using the MST store.

Reply:
If I were to add RxJS back in, I think this time I'd start by subscribing to the stream in the event handlers in the drawing component, then rendering directly from the streamed coordinates without involving the store.

Reply:
I've now run into the same problem, on #1266, where updating a mark does not trigger a new render of the `DrawingContainer` component.